### PR TITLE
Modernise Just Track

### PR DESCRIPTION
**Pull request by danlages**
This change to Just Track consists of the following changes: 

- [x] The Removal of Objective-C Code
- [x] General Refactor (removing Prefixes and the deprecated synchronize calls) 
- [x] The removal of legacy unadopted files and frameworks
- [x] Resolves naming generation to ensure event and payload names adhere to camelCase formatting.
- [x] An update to the generator serving to preserve order accross runs
- [x] An update to the README detailing information about the change
- [x] Facilitate the implementation of an array of objects as part of the payload.

This update will introduce breaking changes to Objective-C users and will also a require an update to the naming of all current events, removing the 'JE' prefix.

### CODE REVIEW

File: JustTrack/EventsGenerator/EventListTemplate.jet [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 /*example
 
-@objcMembers public class JEEventExample: NSObject, JEEvent {
+public class EventExample: NSObject, Event {
```
Comment (albertodebortoli) [inline JustTrack/EventsGenerator/EventListTemplate.jet]: Do we still need to inherit from NSObject?

Comment (danlages) [reply JustTrack/EventsGenerator/EventListTemplate.jet]: Let me research this.

**Review by albertodebortoli — APPROVED** [review]

File: JustTrack/EventsGenerator/EventListTemplate.jet (hunk 1 above)
Comment (danlages) [reply JustTrack/EventsGenerator/EventListTemplate.jet]: Done

File: Example/JustTrack/TrackingEvents.swift [hunk 2]
```diff
@@ -75,41 +80,65 @@ public class EventViewScreen: Event {
         self.screenDataVarSetting = screenDataVarSetting
     }
 }
-
 public class EventExample: Event {
+
     public let name: String = "example"
+    
+    public struct TestObject: Equatable {
+        public var item_name: String = ""
```
Comment (antoniostrijdom) [inline Example/JustTrack/TrackingEvents.swift]: Type inference makes defining the type (e.g. `: String`) redundant.

**Review by antoniostrijdom — CHANGES_REQUESTED** [review]
LGTM. but maybe think about maybe dropping the type from the struct properties.

**Review by albertodebortoli — APPROVED** [review]
Looks good!

File: README.md [hunk 3]
```diff
@@ -165,31 +178,59 @@ An Event is made of:
...
+A Tracker is an object implementing the **Tracker** protocol and is loaded using: ```tracker.loadCustomTracker( ... )``` function. You can implement any tracker you want and **JustTrack** provides a few default trackers:
+
+* [x] TrackerConsole - print events to the system's console
+* [ ] FacebookTraker (not yet implemented)
+* [ ] ~~GoogleAnalyticsTraker~~ (not yet implemented, Google's pods can't be used as a dependency in a pod)
+* [ ] ~~TrakerFirebase~~ (not yet implemented, Google's pods can't be used as a dependency in a pod)
+
+## Upgrading to v4.0
+
+In version 4.0 Just Track has been refactored to bring it up to date with current Swift standards. As such, Just Track no longer supports Objective-C implementations. Please consider updating any applications that consume this feature.  As a result, the way events are named and must be called has been adjusted substatially. Please consider the following points when upgrading.
```
Comment (antoniostrijdom) [inline README.md]: Small thing: best to keep this consistent through the whole readme
```suggestion
In version 4.0 **JustTrack** has been refactored to bring it up to date with current Swift standards. As such, Just Track no longer supports Objective-C implementations. Please consider updating any applications that consume this feature.  As a result, the way events are named and must be called has been adjusted substatially. Please consider the following points when upgrading.
```

**Review by antoniostrijdom — APPROVED** [review]
Nice one.

**Pull request merged**